backend_python/backend/game.py

```diff
@@ -0,0 +1,184 @@
+        self.status = {}
+        self.version = datetime.now().timestamp()
+
+        self.machine = Machine(model=self,
+            states=self.generate_states(),
+            transitions=self.generate_transitions(),
+            initial=Game.initial_state)
+
+    def generate_states(self):
+
+        states = []
+        states.append(State('waiting_for_players', on_enter=self.stamp_state))
+
+        states.append(State('day_voting', 
+            on_enter=[self.stamp_state, self.clear_results],
+            on_exit=self.execute_vote))
+
+        states.append(State('day_results',
+            on_enter=self.stamp_state))
+
+        states.append(State('night_voting', 
+            on_enter=[self.stamp_state, self.clear_results],
+            on_exit=self.execute_vote))
+
+        states.append(State('night_results',
+            on_enter=self.stamp_state))
+            
+        states.append(State('game_ended', 
+            on_enter=[self.stamp_state, self.set_winner]))
+                
+        return states
+
+
+    def generate_transitions(self):
+        transitions = []
+        transitions.append({
+                'trigger': 't_start_game',
+                'source': 'waiting_for_players',
+                'dest': 'day_voting',
+                'conditions': self.town.check_ready_to_start,
+            })
+        transitions.append({
+                'trigger': 't_execute_vote',
+                'source': 'day_voting',
+                'dest': 'day_results',
+                'conditions': None,
+                'after': self.end_game_maybe,
+            })
+        transitions.append({
+                'trigger': 't_progress',
+                'source': 'day_results',
+                'dest': 'night_voting',
+                'conditions': None,
+                'before': self.progress,
+            })
+        transitions.append({
+                'trigger': 't_progress',
+                'source': 'night_voting',
+                'dest': 'night_results',
+                'conditions': self.can_game_continue,
+                'after': self.end_game_maybe, 
+            })
+        transitions.append({
+                'trigger': 't_progress',
+                'source': 'night_results',
+                'dest': 'day_voting',
+                'conditions': None,
+                'before': self.progress,
+            })
+        transitions.append({
+                'trigger': 't_end_game',
+                'source': ['day_results', 'night_results'],
+                'dest': 'game_ended',
+                'before': self.end_game
+            })
+        
+        return transitions
+
+
+    def to_dict(self):
+        dictionary = self.town.to_dict()
+        self.status['id'] = self.state
+        dictionary['state'] = self.status
+        dictionary['version'] = self.version
+
+        return dictionary
+
+    def jversonify(self):
+        self.version = datetime.now().timestamp()
+
+        return(jsonify(self.to_dict()))
+
+
```

Comment:
is there some convention on how many lines should be used to separate methods? maybe we should use https://www.pylint.org/ at some point?